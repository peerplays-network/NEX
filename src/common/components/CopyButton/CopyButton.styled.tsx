import { Button, styled } from "../../../ui/src";
import { Check as UiCheck } from "../../../ui/src/icons";

export const Check = styled(UiCheck)`
  color: #11b881;
`;

export const TextButton = styled(Button)`
  height: unset;
  padding: 0;
`;
