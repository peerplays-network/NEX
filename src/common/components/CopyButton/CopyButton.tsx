import { Tooltip } from "../../../ui/src";
import { CopyIcon } from "../../../ui/src/icons";

import * as Styled from "./CopyButton.styled";
import { useCopyButton } from "./hooks";

type Props = {
  buttonText?: string;
  copyValue: string;
  className?: string;
};

export const CopyButton = ({
  buttonText,
  copyValue,
  className,
}: Props): JSX.Element => {
  const { copied, handleClick } = useCopyButton();

  return (
    <span className={className}>
      {buttonText && buttonText !== "" ? (
        !copied ? (
          <Tooltip placement="top" title="Copy">
            <Styled.TextButton
              type="link"
              onClick={() => handleClick(copyValue)}
            >
              {`${buttonText} `} <CopyIcon />
            </Styled.TextButton>
          </Tooltip>
        ) : (
          <Tooltip placement="top" title="Copied">
            <Styled.TextButton
              type="link"
              onClick={() => handleClick(copyValue)}
            >
              {`${buttonText} `} <Styled.Check />
            </Styled.TextButton>
          </Tooltip>
        )
      ) : !copied ? (
        <Tooltip placement="top" title="Copy">
          <CopyIcon onClick={() => handleClick(copyValue)} />
        </Tooltip>
      ) : (
        <Tooltip placement="top" title="Copied">
          <Styled.Check />
        </Tooltip>
      )}
    </span>
  );
};
