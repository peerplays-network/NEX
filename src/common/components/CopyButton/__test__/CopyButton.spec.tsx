import { render } from "@testing-library/react";
import React from "react";

import { CopyButton } from "../CopyButton";

describe("CopyButton", () => {
  it("should render with button text", () => {
    const { asFragment } = render(
      <CopyButton buttonText="Copy" copyValue="test value" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
