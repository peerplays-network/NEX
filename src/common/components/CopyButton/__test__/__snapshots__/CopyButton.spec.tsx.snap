// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyButton should render with button text 1`] = `
<DocumentFragment>
  <span>
    <button
      class="ant-btn ant-btn-link Button-sc-d40bd577-0 sc-gswNZR dPgDWt fUKfuq"
      type="button"
    >
      <span>
        Copy  
      </span>
      <span
        class="anticon"
        role="img"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
          />
        </svg>
      </span>
    </button>
  </span>
</DocumentFragment>
`;
